linera-views/src/lru_prefix_cache.rs

```diff
@@ -820,6 +820,28 @@ impl LruPrefixCache {
         }
     }
 
+    /// Returns whether the cache can decide if a key is present or not.
+    pub(crate) fn test_key_presence(&self, key: &[u8]) -> bool {
+        // First, query the value map
+        let result = match self.value_map.get(key) {
+            None => false,
+            Some(entry) => match entry {
+                ValueEntry::DoesNotExist => false,
+                ValueEntry::Exists => false,
+                ValueEntry::Value(_) => true,
```

Comment:
maybe just return early here `return true`, other variants no-op. You don't need the intermediate `result` value.

Reply:
The whole function is basically:

```rust
matches!(self.value_map.get(key), Some(ValueEntry::Value(_)))
    || (self.has_exclusive_access && self.get_existing_find_key_values_entry(key).is_some())
```

(But I'm not against a more verbose version, if you find it more readable.)